File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    cnt = 0
    s = list(s)
    for i in range(len(s)):
        if s[i] == "(":
            cnt += 1
        else:
            cnt -= 1
        if cnt < 0:
            return False
    if cnt == 0:
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var flag: Int = 0
    var ans:Bool = true
    for i in s {
        if i == "(" {
            flag += 1
        } else if i == ")" {
            if flag <= 0 {
                return false
            }
            flag -= 1
        }
    }
    
    if flag > 0 {
        ans = false
    }
    return ans
}